src/data/polynomial/degree/definitions.lean

```diff
@@ -1156,6 +1156,20 @@ else if hq0 : q = 0 then  by simp only [hq0, degree_zero, mul_zero, with_bot.add
 else degree_mul' $ mul_ne_zero (mt leading_coeff_eq_zero.1 hp0)
     (mt leading_coeff_eq_zero.1 hq0)
 
+lemma degree_le_of_dvd : (q ≠ 0) → (p ∣ q) → degree p ≤ degree q :=
+begin
+  intros hqnz hpq,
+  cases hpq with g hqpg,
+  exact if hp0: p = 0 then by { rw hp0 at hqpg, rw zero_mul at hqpg, rw [ hp0, hqpg], norm_num, }
```

Comment:
The usual tactic for this is `by_cases hp0 : p = 0`, and then you can have two curly brace blocks following it for each case.  It's rare that `if`s are needed inside lemmas.

Reply:
Used the first one. I don't understand the notation in the rcases